src/docs/markdown/build.md

```diff
@@ -50,24 +72,45 @@ To build with plugins, use `--with`:
 
 As you can see, you can customize the versions of plugins with `@` syntax. Versions can be a tag name, commit SHA, or branch.
 
-Cross-platform compilation with `xcaddy` works the same as with the `go` command (see below).
+Cross-platform compilation with `xcaddy` works the same as with the `go` command. For example, to cross-compile for macOS:
 
+<pre><code class="cmd bash">GOOS=darwin xcaddy build</code></pre>
 
-## Cross-platform
+## Build Custom Docker Image
 
-Go programs are easy to compile for other platforms. Just set the `GOOS`, `GOARCH`, and/or `GOARM` environment variables that are different. ([See the go documentation for details.](https://golang.org/doc/install/source#environment))
+Most users deploying production sites will not want to rely on mounting files into a container, but will instead base their own images on `caddy`:
 
-For example, to compile Caddy for Windows when you're not on Windows:
+```Dockerfile
+# note: never use the :latest tag in a production site
+FROM caddy:<version>
 
-<pre><code class="cmd bash">GOOS=windows go build</code></pre>
+COPY Caddyfile /etc/caddy/Caddyfile
+COPY site /srv
+```
 
-Or similarly for Linux ARMv6 when you're not on Linux or on ARMv6:
+#### Adding custom Caddy modules
 
-<pre><code class="cmd bash">GOOS=linux GOARCH=arm GOARM=6 go build</code></pre>
+Caddy is extendable through the use of "modules". See https://caddyserver.com/docs/extending-caddy for full details. You can find a list of available modules on [the Caddy website's download page](https://caddyserver.com/download).
```

Comment:
I think this line belongs elsewhere than this section, it only makes sense in context of the Docker hub docs which needs to give some background/preamble